services/dynamic-scheduler/src/simcore_service_dynamic_scheduler/services/service_tracker/_api.py

```diff
@@ -0,0 +1,249 @@
+
+    model: TrackedServiceModel | None = await tracker.load(node_id)
+    if model is not None:
+        model.dynamic_service_start = dynamic_service_start
+        model.requested_state = UserRequestedState.RUNNING
+        model.project_id = dynamic_service_start.project_id
+        model.user_id = dynamic_service_start.user_id
+
+    await tracker.save(
+        node_id,
+        TrackedServiceModel(
+            dynamic_service_start=dynamic_service_start,
+            requested_state=UserRequestedState.RUNNING,
+            project_id=dynamic_service_start.project_id,
+            user_id=dynamic_service_start.user_id,
+        ),
+    )
+
+
+async def set_request_as_stopped(
+    app: FastAPI, dynamic_service_stop: DynamicServiceStop
+) -> None:
+    """Stores the intention of the user: ``stop`` requested"""
+    tracker: Tracker = get_tracker(app)
+    model: TrackedServiceModel | None = await tracker.load(dynamic_service_stop.node_id)
+
+    if model is None:
+        model = TrackedServiceModel(
+            dynamic_service_start=None,
+            user_id=dynamic_service_stop.user_id,
+            project_id=dynamic_service_stop.project_id,
+            requested_state=UserRequestedState.STOPPED,
+        )
+
+    model.requested_state = UserRequestedState.STOPPED
+    await tracker.save(dynamic_service_stop.node_id, model)
+
+
+def __get_state_str(status: NodeGet | DynamicServiceGet | NodeGetIdle) -> str:
+    # Attributes where to find the state
+    # NodeGet -> service_state
+    # DynamicServiceGet -> state
+    # NodeGetIdle -> service_state
+    state_key = "state" if isinstance(status, DynamicServiceGet) else "service_state"
+
+    state: ServiceState | str = getattr(status, state_key)
+    result: str = state.value if isinstance(state, ServiceState) else state
+    return result
+
+
+def _get_poll_interval(status: NodeGet | DynamicServiceGet | NodeGetIdle) -> timedelta:
+    if __get_state_str(status) != "running":
+        return _LOW_RATE_POLL_INTERVAL
+
+    return NORMAL_RATE_POLL_INTERVAL
+
+
+def _get_current_state(
+    requested_sate: UserRequestedState,
+    status: NodeGet | DynamicServiceGet | NodeGetIdle,
+) -> SchedulerServiceState:
+    """
+    Computes the `SchedulerServiceState` used internally by the scheduler
+    to decide about a service's future.
+    """
+
+    if isinstance(status, NodeGetIdle):
+        return SchedulerServiceState.IDLE
+
+    service_state: ServiceState = ServiceState(__get_state_str(status))
+
+    if requested_sate == UserRequestedState.RUNNING:
+        if service_state == ServiceState.RUNNING:
+            return SchedulerServiceState.RUNNING
+
+        if (
+            ServiceState.PENDING  # type:ignore[operator]
+            <= service_state
+            <= ServiceState.STARTING
+        ):
+            return SchedulerServiceState.STARTING
+
+        if service_state < ServiceState.PENDING or service_state > ServiceState.RUNNING:
```

Comment:
hmm... is that why you changed the order in ServiceState??? is this not a little bit fragile? I mean the definition is somewhere in models or service lib right? do you have a test that checks it? or better, maybe you would need to explicitely set the order somewhere